src/libcore/slice.rs

```diff
@@ -523,8 +523,7 @@ impl<T> SliceExt for [T] {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-#[allow(unused_attributes)]
-#[rustc_on_unimplemented = "a usize is required to index into a slice"]
+#[rustc_on_unimplemented = "slice indexes must be usizes"]
```

Comment:
Just to make sure the message is as clear as possible to the users that will be helped, maybe we can change to **“slice index type must be usize”**? I just think the “usizes” formulation will be confusing.